api/src/Handler/RBACInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*.
+    require_module 'standard';
+.*/
+
+namespace App\Handler;
+
+interface RBACInterface
```

Comment:
I keep talking myslef into and out of this.

Part of it is that they are interfaces that are talking out of different sides.  

`RBACInterface` talks between the RBAC implementation in into the API
`RBACVendor` talks between the RBAC implementation into the platform backend.  (Zend) 

MANY of the functions pass all the way through from the API -> Staff  RBAC(API Facing RBAC) -> Zend RBAC.

It would be cleaner to have a common interface that API interface and the Vendor interface both inherited to but I am not sure if it make logical sense to me and I worry it would be confusing.

Given that what are your thoughts?

Reply:
On one hand I agree, I think it could be confusing because they are meant for the different interaction areas of the application. On the other, I think about situations where I have done things like this myself, and then make a change in one interface and then do it slightly differently in the other (think swapped parameters or something of that nature) that only makes it even more confusing.

I think it might be less confusing to have a common interface because the interaction from where to where may not be as important as the actual purpose of the function. The API wants to add a permission to a role, and the backend says "OK I will add a permission to a role". In that regard it feels like the behavior is intended to be the same in both places, even though the API implementation is saying "Hey RBAC service, can you take care of this for me?"

I could see the argument for either side of this, but I would tend to lean towards the "Don't Repeat Yourself" mantra of coding, though maybe not until I found myself making the same interface in 3 places (unlikely, but you never know). But I also have a good feeling that if left separated, I might end up in a situation where I see this again in 6 months and go "What was the reason for our approach with this again?"

I'll let you decide on a final verdict, it seems like it will work fine as-is, so it's not a blocker for me.